src/mlpack/methods/ann/modelParser.cpp

```diff
@@ -0,0 +1,1132 @@
+    */
+   void setValidSet(arma::mat& validX, arma::mat& validY)
+   {
+     this->validX = validX;
+     this->validY = validY;
+   }
+   /**
+    * Return the values of the training input dataset
+    */
+   arma::mat getTrainX()
+   {
+     return trainX;
+   }
+   /**
+    * Return the values of the training output dataset
+    */
+   arma::mat getTrainY()
+   {
+     return trainY;
+   }
+   /**
+    * Return the values of the validation input dataset
+    */
+   arma::mat getValidX()
+   {
+     return validX;
+   }
+   /**
+    * Return the values of the validation output dataset
+    */
+   arma::mat getValidY()
+   {
+     return validY;
+   }
+};
+
+/**
+ * Print the given stl map where the keys are of type string and
+ * values are of type double
+ * 
+ * @param params The map to be printed
+ */
+void printMap(map<string, double> params)
+{
+  map<string, double>::iterator itr;
+  for (itr = params.begin(); itr != params.end(); ++itr)
+  {
+    cout << itr->first << " : " << itr->second << endl;
+  }
+}
+
+/**
+ * Update the values of a given stl map with that of another map 
+ * corresponding to the keys that are common
+ * 
+ * Keys are of type string and values are of type double
+ * 
+ * @param origParams The map whose values will be updated
+ * @param newParams The map whose values will be used to update origParams
+ */
+void updateParams(map<string, double> &origParams, map<string, double> &newParams)
+{
+  map<string, double>::iterator itr;
+  for (itr = newParams.begin(); itr != newParams.end(); ++itr)
+  {
+    map<string, double>::iterator itr2 = origParams.find(itr->first);
+    itr2->second = newParams.at(itr->first);
+  }
+  //printMap(origParams);
+}
+
+/**
+ * @author Eugene Freyman
+ * Returns labels bases on predicted probability (or log of probability)
+ * of classes.
+ * @param predOut matrix contains probabilities (or log of probability) of
+ * classes. Each row corresponds to a certain class, each column corresponds
+ * to a data point.
+ * @return a row vector of data point's classes. The classes starts from 1 to
+ * the number of rows in input matrix.
+ */
+arma::Row<size_t> getLabels(const arma::mat& predOut)
+{
+  arma::Row<size_t> pred(predOut.n_cols);
+
+  // Class of a j-th data point is chosen to be the one with maximum value
+  // in j-th column plus 1 (since column's elements are numbered from 0).
+  for (size_t j = 0; j < predOut.n_cols; ++j)
+  {
+    pred(j) = arma::as_scalar(arma::find(
+        arma::max(predOut.col(j)) == predOut.col(j), 1)) + 1;
+  }
+
+  return pred;
+}
+
+/**
+ * @author Eugene Freyman
+ * Returns the accuracy (percentage of correct answers).
+ * @param predLabels predicted labels of data points.
+ * @param realY real labels (they are double because we usually read them from
+ * CSV file that contain many other double values).
+ * @return percentage of correct answers.
+ */
+double accuracy(arma::Row<size_t> predLabels, const arma::mat& realY)
```

Comment:
It is better to include the file instead of writing it again

Reply:
Thanks for the suggestion, but the file is in the *models* repo, so I cannot really include it. Moreover it is basic and simple code, so I am not sure if code repetition can be a big issue here.